marketdata/charts.go

```diff
@@ -28,14 +29,34 @@ func InitCharts() *Charts {
 
 func (c *Charts) GetChartData(coin uint, token string, currency string, timeStart int64) (blockatlas.ChartData, error) {
 	chartsData := blockatlas.ChartData{}
-	for _, c := range c.chartProviders {
-		charts, err := c.GetChartData(coin, token, currency, timeStart)
-		if err != nil {
-			continue
-		}
-		return charts, nil
+	chartsDataChan := make(chan blockatlas.ChartData, len(c.chartProviders))
+	errChan := make(chan struct{})
+
+	var errCount int32
+	for _, provider := range c.chartProviders {
+		go func(provider chart.Provider) {
+			charts, err := provider.GetChartData(coin, token, currency, timeStart)
+			if err != nil || len(charts.Prices) == 0 {
+				if int(atomic.LoadInt32(&errCount)) == len(c.chartProviders) - 1 {
+					errChan <- struct{}{}
+					return
```

Comment:
I would remove this return and use `>=` instead of `==` mark

Reply:
I don't think so - we need to send the value to the error channel only when all the providers did not work correctly